cl/search/tests/test_pacer_bulk_fetch.py

```diff
@@ -0,0 +1,640 @@
+        mock_timed_out_cache_key,
+        mock_fetched_cache_key,
+        mock_should_skip,
+    ):
+        """
+        Test that documents already available in CL are skipped
+        (i.e., not enqueued).
+        """
+
+        # Make some documents available
+        docs_available_in_cl = []
+        for court_idx in range(5):
+            court_docs = [
+                d
+                for d in self.docs
+                if d.docket_entry.docket.court_id == f"ca{court_idx}"
+            ]
+            docs_available_in_cl.extend(
+                court_docs[:2]
+            )  # first 2 docs from each court
+
+        RECAPDocument.objects.filter(
+            pk__in=[d.pk for d in docs_available_in_cl]
+        ).update(is_available=True)
+
+        call_command(
+            "pacer_bulk_fetch",
+            testing=True,
+            stage="fetch",
+            username=self.user.username,
+        )
+
+        # Check how many docs were actually fetched
+        expected_unavailable = [
+            d
+            for d in self.docs
+            if d.pk not in [ad.pk for ad in docs_available_in_cl]
+        ]
+        self.assertEqual(
+            mock_fetch_pacer_doc_by_rd.call_count, len(expected_unavailable)
+        )
+
+        # None of the available docs should be in the calls
+        called_args = [
+            args[0] for args, _ in mock_fetch_pacer_doc_by_rd.call_args_list
+        ]
+        for doc in docs_available_in_cl:
+            with self.subTest(doc=doc):
+                self.assertNotIn(doc.pk, called_args)
+
+        _clean_cache_keys(
+            [
+                mock_fetched_cache_key.return_value,
+                mock_timed_out_cache_key.return_value,
+            ]
+        )
+
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.should_skip",
+        return_value=False,
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.docs_to_process_cache_key",
+        return_value="pacer_bulk_fetch.test_rate_limiting.docs_to_process",
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.timed_out_docs_cache_key",
+        return_value="pacer_bulk_fetch.test_rate_limiting.timed_out_docs",
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.CeleryThrottle.maybe_wait"
+    )
+    @patch("cl.search.management.commands.pacer_bulk_fetch.time.sleep")
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.get_or_cache_pacer_cookies"
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.fetch_pacer_doc_by_rd.si"
+    )
+    def test_rate_limiting(
+        self,
+        mock_fetch_pacer_doc_by_rd,
+        mock_pacer_cookies,
+        mock_sleep,
+        mock_maybe_wait,
+        mock_timed_out_cache_key,
+        mock_fetched_cache_key,
+        mock_should_skip,
+    ):
+        """Test that rate limiting triggers sleep calls (mocked)."""
```

Comment:
What is this test intended to check? Is it different from `test_page_count_filtering`?

Reply:
We agreed that this test can be removed.